### DISCUSSION TITLE
Separate Rails and React Repos/deployments

### DISCUSSION THREAD

**Discussion by brmstephenson** [shakacode/react_on_rails#1538] [Q&A] [answered]
When using react_on_rails is there a way to separate the React code into a different repo and deploy the React code to nginx or even a simple s3 bucket? I'd still like to call `react_component(mycomponent, props)` and just tell Rails to get the component from another location. Is that possible?

**Comment by justin808** [comment]
@brmstephenson Anything is possible. Why would this be worth the extra complexity?

You'd only need to set the output path for the server-bundle.js to be in the regular spot. You can have a shell script to keep that synchronized.

**Reply by brmstephenson** [reply]
I'd like to have separation as we scale out our React project. We expect our React project to grow in size and we'd like the separation of concern as well as having React hosted separately. We'd use shakapacker to handle the Rails side but host the React side elsewhere so I'm not sure how relevant the server-bundle.js would be. Would we need Rails to download the React assets or something with a shell script to be able to call `react_component(mycomponent, props)`?

**Comment by Judahmeek** [comment, marked as answer]
Your question isn't really relevant to ReactOnRails.

It's more of a Shakapacker concern & should be able to be resolved with the [SHAKAPACKER_ASSET_HOST](https://github.com/shakacode/shakapacker/blob/4ffb77ed3b741975256d73943be6b373d4d51d08/package/__tests__/config.js#L21) environment variable.
